src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -124,6 +122,68 @@ private DatadogHttpClient(String url, String logIntakeUrl, Secret apiKey) {
         this.logIntakeUrl = logIntakeUrl;
     }
 
+    public static void validateCongiguration(String url, String logIntakeUrl, Secret apiKey) throws RuntimeException {
+        if (url == null || url.isEmpty()) {
+            throw new RuntimeException("Datadog Target URL is not set properly");
+        }
+        if (apiKey == null || Secret.toString(apiKey).isEmpty()){
+            throw new RuntimeException("Datadog API Key is not set properly");
+        }
+        if (DatadogHttpClient.isCollectBuildLogEnabled() && (logIntakeUrl == null || logIntakeUrl.isEmpty())){
+            throw new RuntimeException("Datadog Log Intake URL is not set properly");
+        }
+        try {
+            boolean intakeConnection = validateDefaultIntakeConnection(url, apiKey);
+            if (!intakeConnection) {
+                instance.setDefaultIntakeConnectionBroken(true);
+
+                throw new RuntimeException("Connection broken, please double check both your API URL and Key");
+            }
+        } catch (IOException e) {
+            instance.setDefaultIntakeConnectionBroken(true);
+            throw new RuntimeException("Connection broken, please double check both your API URL and Key");
+        }
+        try {
+            boolean logConnection = validateLogIntakeConnection(url, apiKey);
+            if (!logConnection) {
+                instance.setLogIntakeConnectionBroken(true);
+                throw new RuntimeException("Connection broken, please double check both your Log Intake URL and Key");
+            }
+        } catch (IOException e) {
+            instance.setLogIntakeConnectionBroken(true);
+            throw new RuntimeException("Connection broken, please double check both your Log Intake URL and Key");
+        }
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (!(object instanceof DatadogClient)) {
+            return false;
+        }
+
+        DatadogHttpClient newInstance = (DatadogHttpClient) object;
+        DatadogHttpClient httpInstance = (DatadogHttpClient) instance;
+
+        if ((StringUtils.equals(httpInstance.getLogIntakeUrl(), newInstance.getLogIntakeUrl()))
+        && (StringUtils.equals(httpInstance.getUrl(), newInstance.getUrl())
+        && ((newInstance.getApiKey() == null && httpInstance.getApiKey() == null) || newInstance.getApiKey().equals(httpInstance.getApiKey())))){
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Auto generated ? Let's indicate it

Reply:
I based it off of the method from our other `hashCode` methods; it was recommended to override `hashCode` since I overrode `equals`